mailet/standard/src/test/java/org/apache/james/transport/matchers/AtMostTest.java

```diff
@@ -0,0 +1,229 @@
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.transport.matchers;
+
+import static org.apache.mailet.base.MailAddressFixture.RECIPIENT1;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.Collection;
+
+import javax.mail.MessagingException;
+
+import org.apache.james.core.MailAddress;
+import org.apache.mailet.Attribute;
+import org.apache.mailet.AttributeName;
+import org.apache.mailet.AttributeValue;
+import org.apache.mailet.Mail;
+import org.apache.mailet.base.test.FakeMail;
+import org.apache.mailet.base.test.FakeMatcherConfig;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+class AtMostTest {
+    private static final String NO_VALUE_MATCHER = "NoValueMatcher";
+    private static final String RETRIES_CONDITION = "2";
+    private static final AttributeName AT_MOST_RETRIES = AttributeName.of(RETRIES_CONDITION);
+    private static final String NOT_MATTER = "noMatter";
+    private static final String RETRY_5_TIMES = "5:" + NOT_MATTER;
+    private static final String RETRY_3_TIMES = "3:" + NOT_MATTER;
+    private static final String RETRY_WITHOUT_CONDITION_NAME = "3:";
+    private static final String RETRY_WITHOUT_CONDITION_VALUE = ":" + NOT_MATTER;
+    private static final String RETRY_WITH_SPACE_IN_CONDITION = "  :  ";
+
+    private AtMost matcher;
+    private AtMost matchers;
+
+    private Mail createMail() throws MessagingException {
+        return FakeMail.builder()
+            .name("test-message")
+            .recipient(RECIPIENT1)
+            .build();
+    }
+
+    @BeforeEach
+    void setup() throws MessagingException {
+        this.matcher = new AtMost();
+        FakeMatcherConfig matcherConfig = FakeMatcherConfig.builder()
+            .matcherName("SingleMatcher")
+            .condition(RETRIES_CONDITION)
+            .build();
+        matcher.init(matcherConfig);
+
+        this.matchers = new AtMost();
+        matchers.init(
+            FakeMatcherConfig.builder()
+                .matcherName("MultipleMatchers")
+                .condition(RETRY_3_TIMES)
+                .condition(RETRY_5_TIMES)
+                .build());
```

Comment:
I suggest to write two nested class, each one setting up their own matcher.

One using the default attribute name, the other one using a custom attribute name.